pyproject-fmt/rust/src/dependency_groups.rs

```diff
@@ -0,0 +1,29 @@
+use common::array::{sort, transform};
+use common::pep508::{format_requirement, get_canonic_requirement_name};
+use common::table::{collapse_sub_tables, for_entries, reorder_table_keys, Tables};
+
+#[allow(clippy::too_many_lines)]
+pub fn fix(tables: &mut Tables,
+    keep_full_version: bool,) {
+    collapse_sub_tables(tables, "dependency-groups");
+    let table_element = tables.get("dependency-groups");
+    if table_element.is_none() {
+        return;
+    }
+    let table = &mut table_element.unwrap().first().unwrap().borrow_mut();
+    for_entries(table, &mut |_key, entry| {
+        transform(entry, &|s| format_requirement(s, keep_full_version));
+        sort(entry, |e| {
+            get_canonic_requirement_name(e).to_lowercase() + " " + &format_requirement(e, keep_full_version)
+        });
+    });
+    reorder_table_keys(
+        table,
+        &[
+            "",
+            "dev",
+            "test",
+            "docs",
+        ],
+    );
```

Comment:
Was much easier to find an example of this kind of ordering than alphabetical... I thought it might make sense to order group by order of "importance":

- `dev` first, as it has a special meaning with (at least) Poetry & uv, probably others too
- `test` as I think I've seen it as a commonly used group in `requirements_test.txt`
- `docs` feels also relatively common but less important in the dev workflow than the other 2...

Let me know your thoughts, it's a bit arbitrary.

On the long run, perhaps alphabetical would be easier for you to say no to adding more arbitrary rules to this ordering?

Reply:
I am happy with your suggestion.